source/state_representation/include/state_representation/trajectory/TrajectoryBase.hpp

```diff
@@ -0,0 +1,240 @@
+  std::chrono::nanoseconds duration;
+};
+
+template<typename TrajectoryT>
+class TrajectoryBase : public State {
+public:
+  /**
+   * @brief Get attribute list of trajectory point durations
+   * @return the list of trajectory point durations
+   */
+  const std::deque<std::chrono::nanoseconds> get_durations() const;
+
+  /**
+   * @brief Get attribute list of trajectory point times from start
+   * @return the list of trajectory point times from start
+   */
+  const std::deque<std::chrono::nanoseconds> get_times_from_start() const;
+
+  /**
+   * @brief Get attribute number of point in trajectory
+   * @return the number of points in trajectory
+   */
+  unsigned int get_size() const;
+
+  /**
+   * @brief Reset trajectory
+   */
+  virtual void reset();
+
+  /**
+   * @brief Delete last point and corresponding time from trajectory
+   */
+  virtual void delete_point();
+
+  /**
+   * @brief Clear trajectory
+   */
+  virtual void clear();
+
+protected:
+  /**
+   * @brief Empty constructor
+   */
+  explicit TrajectoryBase();
+
+  /**
+   * @brief Constructor with name provided
+   * @param name the name of the state
+   */
+  explicit TrajectoryBase(const std::string& name);
+
+  /**
+   * @brief Get attribute list of trajectory points
+   * @return the list of trajectory points
+   */
+  const std::deque<TrajectoryT>& get_points() const;
+
+  /**
+   * @brief Get the trajectory point at given index
+   * @param index the index
+   * @return the trajectory point
+   * @throw std::out_of_range if pos is out of range
+   */
+  const TrajectoryT& get_point(unsigned int index) const;
+
+  /**
+   * @brief Add new point to trajectory
+   * @param new_point the new point
+   */
+  void add_point(const TrajectoryT& new_point);
+
+  /**
+   * @brief Insert new trajectory point between two already existing points
+   * @param new_point the new point
+   * @param pos the desired position of the new point in the queue
+   * @throw std::out_of_range if pos is out of range
+   */
+  void insert_point(const TrajectoryT& new_point, unsigned int pos);
+
+  /**
+   * @brief Get the trajectory point at given index
+   * @param index the index
+   * @return the trajectory point
+   */
+  TrajectoryT& get_point(unsigned int index);
+
+  /**
+   * @brief Set the trajectory point at given index
+   * @param point the new point
+   * @param index the index
+   * @throw std::out_of_range if index is out of range
+   */
+  void set_point(const TrajectoryT& point, unsigned int index);
+
+  /**
+   * @brief Set the trajectory points from a vector of points
+   * @param points vector of new points
+   */
+  void set_points(const std::vector<TrajectoryT>& points);
+
+  /**
+   * @brief Operator overload for returning a single trajectory point and
+   * corresponding time
+   * @param idx the index
+   * @return the trajectory point
+   * @throw std::out_of_range if index is out of range
+   */
+  const TrajectoryT operator[](unsigned int idx) const;
```

Comment:
```suggestion
  const TrajectoryT& operator[](unsigned int idx) const;
```
or not? Why would you want to return a const object that is not a reference?